File: tbd-p2-backend/src/main/java/com/tbd/bank_backend/models/Transaction.java
package com.tbd.bank_backend.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "transactions")
public class Transaction {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int id;
	@ManyToOne
	@JoinColumn(name = "account_id")
	Account account;
	@Column(nullable = false)
	double amount;
	@ManyToOne
	@JoinColumn(name = "type")
	TransactionType type;
	@ManyToOne
	@JoinColumn(name = "status")
	TransactionStatus status;
	String category;
	String description;
	LocalDate date;
	String merchantName;
}

File: tbd-p2-frontend/src/app/models/Transaction.ts
import { Account } from "./Account";
import { TransactionStatus } from "./TransactionStatus";
import { TransactionType } from "./TransactionType";

export interface Transaction {
    id?: number;
    account?: Account;
    amount: number;
    type: TransactionType;
    status: TransactionStatus;
    category: string;
    description: string;
    date?: number;
    merchantName: string;
}